### PR TITLE
[Macros] Don't invoke `ResolveMacroRequest` for the same custom attribute with different `DeclContext`s.

### PR DESCRIPTION
**Pull request by hborla**
Otherwise, double type checking of macro arguments ensues.

### CODE REVIEW

**Comment by hborla** [comment]
@swift-ci please smoke test

**Comment by hborla** [comment]
@swift-ci please build toolchain

File: lib/AST/Decl.cpp [hunk 1]
```diff
@@ -419,6 +413,13 @@ void Decl::forEachAttachedMacro(MacroRole role,
   }
 }
 
+MacroDecl *Decl::getResolvedMacro(CustomAttr *customAttr) const {
```
Comment (hborla) [inline lib/AST/Decl.cpp:416]: I suppose you can still mess this up by calling it on the wrong `Decl`. Perhaps there should be an assertion that `customAttr` appears in the decl's attribute list...

Comment (DougGregor) [reply lib/AST/Decl.cpp:416]: That seems unlikely enough that I'm not worried about it. Some day I do want to get a ConcreteDeclRef out of this query so we have the substitution map for generic macros.

**Review by DougGregor — APPROVED** [review]

**Pull request merged**